aea/cli/add.py

```diff
@@ -181,58 +155,124 @@ def _add_item(click_context, item_type, item_public_id) -> None:
             item_type, public_id=item_public_id, cwd=ctx.cwd, dest=dest_path
         )
 
+    item_config = _load_item_config(item_type, package_path)
+
+    if not _is_fingerprint_correct(package_path, item_config):
+        raise click.ClickException("Failed to add an item with incorrect fingerprint.")
+
+    _add_item_deps(click_context, item_type, item_config)
+    _register_item(ctx, item_type, item_public_id)
+
+
+def _is_item_present(ctx: Context, item_type: str, item_public_id: PublicId) -> bool:
+    """
+    Check if item is already present in AEA.
+
+    :param ctx: context object.
+    :param item_type: type of an item.
+    :param item_public_id: PublicId of an item.
+
+    :return: boolean is item present.
+    """
+    item_type_plural = item_type + "s"
+    dest_path = Path(
+        ctx.cwd, "vendor", item_public_id.author, item_type_plural, item_public_id.name
+    )
+    # check item presence only by author/package_name pair, without version.
+    items_in_config = set(
+        map(lambda x: (x.author, x.name), getattr(ctx.agent_config, item_type_plural))
+    )
+    return (
+        item_public_id.author,
+        item_public_id.name,
+    ) in items_in_config and dest_path.exists()
+
+
+def _add_protocols(
+    click_context: ClickContext, protocols: Collection[PublicId]
+) -> None:
+    """
+    Add protocols to AEA by list of public IDs.
+
+    :param click_context: click context object.
+    :param protocols: a Collection of protocol public IDs to be added.
+
+    :return: None
+    """
+    ctx = cast(Context, click_context.obj)
+    for protocol_public_id in protocols:
+        if protocol_public_id not in ctx.agent_config.protocols:
```

Comment:
Just move these two lines to 253, no need for an extra method..confuses more than it helps

Reply:
Agree. Basically no need as we already check that in _is_item_present.